src/java.desktop/share/classes/javax/swing/plaf/metal/MetalBorders.java

```diff
@@ -241,41 +274,97 @@ public void paintBorder(Component c, Graphics g, int x, int y,
-                  g.setColor(highlight);
-                  // Draw the Long highlight lines
-                  g.drawLine( corner+1, 3, w-corner, 3);
-                  g.drawLine( 3, corner+1, 3, h-corner);
-                  g.drawLine( w-2, corner+1, w-2, h-corner);
-                  g.drawLine( corner+1, h-2, w-corner, h-2);
-
-                  g.setColor(shadow);
-                  // Draw the Long shadow lines
-                  g.drawLine( corner, 2, w-corner-1, 2);
-                  g.drawLine( 2, corner, 2, h-corner-1);
-                  g.drawLine( w-3, corner, w-3, h-corner-1);
-                  g.drawLine( corner, h-3, w-corner-1, h-3);
-              }
+            if (resetTransform) {
+                width = roundDown(at.getScaleX() * w);
+                height = roundDown(at.getScaleY() * h);
+                xtranslation = roundDown(at.getScaleX() * x + at.getTranslateX());
+                ytranslation = roundDown(at.getScaleY() * y + at.getTranslateY());
+            } else {
+                width = w;
+                height = h;
+                xtranslation = x;
+                ytranslation = y;
+            }
+            g2d.translate(xtranslation, ytranslation);
 
-          }
+            // border and corner scaling
+            int scaledCorner = (int) Math.round(corner * at.getScaleX());
+            // loop constraint for bulk of the border
+            int loopCount = (int) Math.round(5 * at.getScaleX());
 
-          public Insets getBorderInsets(Component c, Insets newInsets) {
-              newInsets.set(5, 5, 5, 5);
-              return newInsets;
-          }
+            // midpoint at which highlight & shadow lines
+            // are positioned on the border
+            int midPoint = loopCount/2;
+
+            g.setColor(background);
+            // Draw outermost lines
+            g.drawLine( 0, 0, width-1, 0);
+            g.drawLine( 0, 1, 0, height-1);
+            g.drawLine( width-1, 1, width-1, height-1);
+            g.drawLine( 1, height-1, width-1, height-1);
+
+            // Draw the bulk of the border
+            for (int i = 1; i <= loopCount; i++) {
+                g.drawRect(x+i,y+i,width-(i*2)-1, height-(i*2)-1);
+            }
+
+            if (c instanceof JInternalFrame && ((JInternalFrame)c).isResizable()) {
+                // Draw the Long highlight lines
+                g.setColor(highlight);
+                g.drawLine(scaledCorner + 1, midPoint+stkWidth,
+                        width- scaledCorner, midPoint+stkWidth); //top
+                g.drawLine(midPoint+stkWidth, scaledCorner + 1,
+                        midPoint+stkWidth, height- scaledCorner); //left
+                g.drawLine(width-midPoint, scaledCorner + 1,
+                        width-midPoint, height- scaledCorner); //right
+                g.drawLine(scaledCorner + 1, height-midPoint,
+                        width- scaledCorner, height-midPoint); //bottom
```

Comment:
This looks a bit weird to me. The old code used no spaces around the binary operations, the new code is inconsistent: some cases do, some don't, some have space only after the operator.

Reply:
Updated